### PR TITLE
ussodischarge: add LegacyInteract implementation

### PR DESCRIPTION
**Pull request by rogpeppe**
This enables this interactor to work with the currently
deployed version of the identity manager.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/idmclient/119/

File: ussodischarge/client.go [hunk 1]
```diff
@@ -79,6 +81,22 @@ func (i *Interactor) Interact(ctx context.Context, client *httpbakery.Client, lo
 	return resp.DischargeToken, errgo.Mask(err)
 }
 
+func (i *Interactor) LegacyInteract(ctx context.Context, client *httpbakery.Client, location string, visitURL *stdurl.URL) error {
```
Comment (frankban) [inline ussodischarge/client.go]: Docstring and test for this new method?

File: idmtest/idmtest.go [hunk 2]
```diff
@@ -92,7 +92,6 @@ func NewServer() *Server {
 		Checker:        checker,
 		IdentityClient: identityClient{srv},
 		Authorizer: bakery.ACLAuthorizer{
-			AllowPublic: true,
 			GetACL:      srv.getACL,
```
Comment (mhilton) [inline idmtest/idmtest.go]: This presumably needs formatting now?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/idmclient/120/

File: ussodischarge/client.go (hunk 1 above)
Comment (rogpeppe) [reply ussodischarge/client.go]: I added a doc comment, but testing it seems like a lot of work for the amount of extra reassurance it'll give us (not much, given that we can't test against the real idm server and that the code is pretty small).

Code has been QA'd - it does actually work.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/idmclient-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/idmclient/121/

**Pull request merged**